packages/picasso/src/LexicalEmojiPlugin/nodes/CustomEmojiNode.tsx

```diff
@@ -0,0 +1,153 @@
+import React from 'react'
+import { $applyNodeReplacement, DecoratorNode } from 'lexical'
+import type {
+  DOMConversionMap,
+  DOMConversionOutput,
+  DOMExportOutput,
+  EditorConfig,
+  LexicalNode,
+  NodeKey,
+  SerializedElementNode,
+  Spread,
+} from 'lexical'
+
+export interface CustomEmojiPayload {
+  src: string
+  id: string
+}
+
+type SerializedCustomEmojiNode = Spread<
+  {
+    src: string
+    id: string
+  },
+  SerializedElementNode
+>
+
+const convertImageElement = (domNode: Node): null | DOMConversionOutput => {
+  if (domNode instanceof HTMLImageElement) {
+    const src = domNode.getAttribute('src')
+    const id = domNode.getAttribute('data-emoji-name')
+
+    if (src && id) {
+      return {
+        node: $createCustomEmojiNode({
+          src,
+          id,
+        }),
+      }
+    }
+
+    return null
+  }
+
+  return null
+}
+
+export class CustomEmojiNode extends DecoratorNode<JSX.Element> {
+  __src: string
+  __id: string
+
+  static getType() {
+    return 'custom-emoji'
+  }
+
+  static clone(node: CustomEmojiNode): CustomEmojiNode {
+    return new CustomEmojiNode(node.__src, node.__id)
+  }
+
+  constructor(src: string, id: string, key?: NodeKey) {
+    super(key)
+    this.__src = src
+    this.__id = id
+  }
```

Comment:
On the previous implementation, we agreed on using Picasso code conventions over Lexical's here, or did something change? We can also use the constructor trick to make the code simpler
```suggestion
  static getType() {
    return 'custom-emoji'
  }

  static clone(node: CustomEmojiNode): CustomEmojiNode {
    return new CustomEmojiNode(node.__src, node.__id)
  }

  constructor(public src: string, public id: string, key?: NodeKey) {
    super(key)
  }
```

Reply:
you are right, I forgot to apply this suggestion 🙇

Reply:
it seems that typescript understands it, but webpack has problem with it.
![image](https://github.com/toptal/picasso/assets/6830426/0a1638c1-5049-4a62-aabf-f2070b2c9d23)
So I at least removed the prefix

Reply:
That's weird, I don't know why would webpack break with that syntax, maybe some limitation with babel-typescript, but I'm not sure. Well, it is at least better now